fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,221 @@
+
+**Table 2 – Data model for Tracing values**
+
+###### Table 3
+
+| Version | Format | Regex | Reference |
+| --- | --- | --- | --- |
+| 00 | {trace-id}_-_{parent-id}_-_{trace-flags} | `[0-9a-f]{32}-[0-9a-f]{16}-[0-9a-f]{2}` | Section _[3.2.2.2 version-format](https://www.w3.org/TR/trace-context-1/#version-format)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#version-format)</sup> |
+
+**Table 3 – Supported version-formats**
+
+###### Table 4
+
+| Name | Cardinality | Type | Description | Reference |
+| --- | --- | --- | --- | --- |
+| list-member-keyvalue | 1 | String(3...512) | Key-value pair joined by `=`, describing _vendor_ specific tracing information.<br/>Format: _{vendor-name}_=_{vendor-value}_<br/>Example: `vendorname1=opaqueValue1`  | Section _[3.3.1.3 list-members](https://www.w3.org/TR/trace-context-1/#list)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#list) |
+| vendor-name | 1 | String(1-256) | Vendor (list-member) identifier which is printable lowercase ASCII which only includes `a-z`, `0-9`, underscores (`_`), dashes (`-`), asterisks (`*`), and forward slashes (`/`). | Section _[3.3.1.3.1 Key](https://www.w3.org/TR/trace-context-1/#key)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#key) |
+| vendor-value | 1 | String(1-256) |  Opaque printable lowercase ASCII string excluding comma (`,`), equals (`=`), tabs, newlines, and carriage returns. | Section _[3.3.1.3.2 Value](https://www.w3.org/TR/trace-context-1/#value)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#value) |
+
+**Table 4 – Data model for _tracestate_ list member values**
+
+### 3.2 Generating a Trace
+
+To create the a trace, the following steps are performed:
+
+3.2.1. Generate the _traceparent_ ([Table 1](#table-1)) header with the following values:
+   > 3.2.1.1. Set the _version_ ([Table 2](#table-2)) to `00`.<br/>
+   > 3.2.1.2. Generate the _trace-id_ ([Table 2](#table-2)) to represent the entire end-to-end trace.<br/>
+   > 3.2.1.3. Generate a _parent-id_ ([Table 2](#table-2)) to represent a distribute portion of the trace.<br/>
+   > 3.2.2.4 Set _trace-flags_ ([Table 2](#table-2)) to `01` to indicate if the trace was sampled by the source, otherwise to `00`.<br/>
+
+3.2.2. Generate the _tracestate_ header, value below is for example purposes only as it is vendor specific: 
+   > 3.2.2.1. Add a _list-member-keyvalue_ ([Table 4](#table-4)) with value `{vendor-name}=BASE64Encoded({parent-id})`, where _vendor-name_ ([Table 4](#table-4)) is the FSP or Hub identifier and _vendor-value_ ([Table 4](#table-4)) being `BASE64Encoded({parent-id})` with the operation to base 64 encode the _parent-id_ ([Table 2](#table-2)). Base64 encoding the list member value is optional but required if its contents is not printable ASCII<sup>[1](https://www.w3.org/TR/trace-context-1/#value)</sup>.
```

Comment:
Is it a requirement to use FSP identifier as vendor name? Should this not be stated in Table 4 then?

> ... where _vendor-name_ ([Table 4](#table-4)) is the FSP or Hub identifier ...